packages/plugin-core/src/WorkspaceWatcher.ts

```diff
@@ -152,7 +152,9 @@ export class WorkspaceWatcher {
 
   async onDidSaveTextDocument(document: TextDocument) {
     if (SchemaUtils.isSchemaUri(document.uri)) {
-      await SchemaSyncService.instance().onDidSave({ document });
+      await ExtensionProvider.getExtension().schemaSyncService.onDidSave({
```

Comment:
Rather than bundling schemaSync functionality into WorkspaceWatcher, can we do one of the following. 
 Either 

1) refactor it to have either _extension.ts or a different class register its own callback with vscode's `onDidSaveTextDocument()` event?  That way we remove another dep. from workspace watcher, and have that class manage its own dispose() / lifecycle events.

2) Or pass in ISchemaSyncSvc into WorkspaceWatcher's constructor. (Or injected via IDendronExtension)

Reply:
> register its own callback with vscode

Thinking about window watcher and workspace watcher, we generally tended to just register one watcher and have it invoke required functionality there. Would you say in general we should avoid that when possible? That should also improve maintainability since we don't have to update this centralized code that's dispatching everything explicitly.

Reply:
> 2. Or pass in ISchemaSyncSvc into WorkspaceWatcher's constructor. (Or injected via IDendronExtension)

Passing ISchemaSyncService to workspace constructor. Since that is much simpler than changing how we do callbacks. Choosing the most simpler routes for interface addition refactors, to keep these changes as simple as possible so we can get to 0 circular dependencies as soon as possible :)

Reply:
@SeriousBug I presume that question is to @jonathanyeung since his reply is quoted.

Reply:
@SeriousBug - great question, and yes, going forward we should avoid when possible.  Registration of a callback should occur within the class owning that functionality when it makes sense.  I'll have more details written up as an addendum to Nickolay's async on circular dependencies.